enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,202 @@
+
+1. User sets `enableMetricsServer: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config`
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: cluster-monitoring-config
+      namespace: openshift-monitoring
+    data:
+      config.yaml: |
+        enableMetricsServer: true 
+    ```
+
+1. Deploy and configure the metrics-server in the `openshift-monitoring` namespace
+1. Verify `metrics-server` resource API is functional by querying metrics-server endpoint (TBD)
+1. Update `APIService` resource to point to metrics-server API instead of `prometheus-adapter` API
+1. Delete/Scale Down `prometheus-adapter` deployment
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+  ```go
+  type ClusterMonitoringConfiguration struct {
+    // Enable MetricsServer
+    EnableMetricsServer bool `json:"enableMetricsServer,omitempty"`
+  }
+  ```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Configuration/Enablement
+
+Metrics Server is not enabled by default. It must be explicitly enabled by the user. It is enabled by setting 
+`enableMetricsServer: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config`
+
+**Note:** In the long run, the existing cluster-monitoring-config which is currently a config-map will be moved to a Custom Resource  and is out of scope for this enhancement.
+
+#### Resource Impact
+
+Once metrics-server is enabled, by default 2 replicas of metrics-server instances are being deployed in the openshift-monitoring namespace.
+
+As of OpenShift 4.13 the following idle resource usage have been measured:
+
+Kubelet: TBD
+Metrics Server: TBD
+
+**TODO:** Capture the details on performance testing done from [MON-3210](https://issues.redhat.com/browse/MON-3210)
+
+
+### Risks and Mitigations
+
+One of the risk is resource impact on larger clusters. We need to do performance testing on larger OpenShift cluster to measure the impact on Kubelet when metrics-server is deployed.
+
+[MON-3210](https://issues.redhat.com/browse/MON-3210) is created for doing this.
+
+### Drawbacks
+
+N/A
+
+
+## Design Details
+
+### Open Questions [optional]
+
+[custom-metrics-autoscaler](https://github.com/openshift/custom-metrics-autoscaler-operator) also deploys metrics-server when user enables it, will this affect metrics-server we deploy in monitoring stack?
+
+### Test Plan
+
+The proposed solution will be tested with two suites:
+
+* Unit tests - running independently of each other and without side effects
+* End to end tests - in the same repository as the cluster-monitoring-operator. These tests will be executed against an actual Kubernetes cluster and are intended to verify end-to-end functionality of the entire system.
+
+### Graduation Criteria
+
+We plan to have the feature released Tech Preview first. We would like to test the performance of metrics-server on larger clusters first
```

Comment:
As we're not using a CRD for CMO configuration, can you expand a bit on our TechPreview approach?